### PR TITLE
feat: enable gosec and default linter set

### PR DESCRIPTION
**Pull request by jpwhitemn**
fixes: #352
Signed-off-by: jpwhitemn <jpwhite_mn@yahoo.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-mqtt-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) no new features added to product; just linting
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) n/a
  <link to docs PR>

### CODE REVIEW

File: internal/driver/driver.go [hunk 1]
```diff
@@ -150,7 +150,11 @@ func (d *Driver) handleReadCommandRequest(req sdkModel.CommandRequest, topic str
 	driver.Logger.Debugf("Parse command response: %v", cmdResponse)
 
 	var response map[string]interface{}
-	json.Unmarshal([]byte(cmdResponse), &response)
+	err = json.Unmarshal([]byte(cmdResponse), &response)
+	if err != nil {
+		driver.Logger.Errorf("Error unmarshaling response: %s", err)
```
Comment (bnevis-i) [inline internal/driver/driver.go:155]: If we really have an error here, shouldn't we be exiting the function early?

File: internal/driver/incominglistener.go [hunk 2]
```diff
@@ -42,8 +42,10 @@ func (d *Driver) onIncomingDataReceived(client mqtt.Client, message mqtt.Message
 		reading = string(message.Payload())
 	} else {
 		var data map[string]interface{}
-		json.Unmarshal(message.Payload(), &data)
-
+		err := json.Unmarshal(message.Payload(), &data)
+		if err != nil {
+			driver.Logger.Errorf("Error unmarshaling payload: %s", err)
```
Comment (bnevis-i) [inline internal/driver/incominglistener.go:47]: If we really have an error here, shouldn't we be exiting the function early?

File: internal/driver/responselistener.go [hunk 3]
```diff
@@ -30,7 +30,10 @@ func (d *Driver) onCommandResponseReceived(client mqtt.Client, message mqtt.Mess
 		var response map[string]interface{}
 		var ok bool
 
-		json.Unmarshal(message.Payload(), &response)
+		err := json.Unmarshal(message.Payload(), &response)
+		if err != nil {
+			driver.Logger.Errorf("Error unmarshaling payload: %s", err)
```
Comment (bnevis-i) [inline internal/driver/responselistener.go:35]: If we really have an error here, shouldn't we be exiting the function early?

File: mock/device.go [hunk 4]
```diff
@@ -108,7 +108,10 @@ func runDataSender() {
 func onCommandReceivedFromBroker(client mqtt.Client, message mqtt.Message) {
 	var request map[string]interface{}
 
-	json.Unmarshal(message.Payload(), &request)
+	err := json.Unmarshal(message.Payload(), &request)
+	if err != nil {
+		log.Println(fmt.Sprintf("Error unmarshaling payload: %s", err))
```
Comment (bnevis-i) [inline mock/device.go:113]: If we really have an error here, shouldn't we be exiting the function early?

**Review by bnevis-i — COMMENTED** [review]
Code is much improved, but would like to see more action taken on errors than just logging.

**Comment by jpwhitemn** [comment]
recheck

**Review by bnevis-i — APPROVED** [review]

**Review by lenny-intel — APPROVED** [review]

**Pull request merged**